dart/dynamics/ShapeFrame.h

```diff
@@ -0,0 +1,372 @@
+
+  /// Destructor
+  virtual ~VisualDataProperties() = default;
+};
+
+struct CollisionDataProperties
+{
+  /// This object is collidable if true
+  bool mCollidable;
+
+  /// Constructor
+  CollisionDataProperties(const bool collidable = true);
+
+  /// Destructor
+  virtual ~CollisionDataProperties() = default;
+};
+
+struct DynamicsDataProperties
+{
+  /// Coefficient of friction
+  double mFrictionCoeff;
+
+  /// Coefficient of restitution
+  double mRestitutionCoeff;
+
+  /// Constructor
+  DynamicsDataProperties(const double frictionCoeff = 1.0,
+                         const double restitutionCoeff = 0.0);
+
+  /// Destructor
+  virtual ~DynamicsDataProperties() = default;
+};
+
+} // namespace detail
+
+class VisualAddon final :
+    public common::AddonWithVersionedProperties<
+        VisualAddon,
+        detail::VisualDataProperties,
+        ShapeFrame>
+{
+public:
+
+  using BaseClass = common::AddonWithVersionedProperties<
+      VisualAddon, detail::VisualDataProperties, ShapeFrame>;
+
+  /// Constructor
+  VisualAddon(common::AddonManager* mgr,
+              const PropertiesData& properties = PropertiesData());
+
+  VisualAddon(const VisualAddon&) = delete;
+
+  /// Set RGBA color
+  void setRGBA(const Eigen::Vector4d& color);
+
+  DART_COMMON_GET_ADDON_PROPERTY( Eigen::Vector4d, RGBA )
+  // void setRGBA(const Eigen::Vector4d& value);
+  // const Eigen::Vector4d& getRGBA() const;
+
+  DART_COMMON_SET_GET_ADDON_PROPERTY( bool, Hidden )
+  // void setHidden(const Eigen::Vector4d& value);
+  // const Eigen::Vector4d& getHidden() const;
+
+  /// Identical to setRGB(const Eigen::Vector3d&)
+  void setColor(const Eigen::Vector3d& color);
+
+  /// Identical to setRGBA(const Eigen::Vector4d&)
+  void setColor(const Eigen::Vector4d& color);
+
+  /// Set RGB color components (leave alpha alone)
+  void setRGB(const Eigen::Vector3d& rgb);
+
+  /// Set the transparency of the Shape
+  void setAlpha(const double alpha);
+
+  /// Get color
+  Eigen::Vector3d getColor() const;
+
+  /// Get RGB color components
+  Eigen::Vector3d getRGB() const;
+
+  /// Get the transparency of the Shape
+  double getAlpha() const;
+
+  /// Hide the ShapeNode
+  void hide();
+
+  /// Show the ShapeNode
+  void show();
+
+  /// True iff the ShapeNode is set to be hidden. Use hide(bool) to change this
+  /// setting
+  bool isHidden() const;
+
+protected:
+
+  /// The parent shape frame
+  ShapeFrame* mShapeFrame;
+
+};
+
+class CollisionAddon final :
+    public common::AddonWithVersionedProperties<
+        CollisionAddon,
+        detail::CollisionDataProperties,
+        ShapeFrame>
+{
+public:
+
+  CollisionAddon(const CollisionAddon &) = delete;
+  CollisionAddon(dart::common::AddonManager* mgr,
+                      const PropertiesData& properties = PropertiesData());
+
+  DART_COMMON_SET_GET_ADDON_PROPERTY( bool, Collidable )
+  // void setCollisionMode(const bool& value);
+  // const bool& getCollisionMode() const;
```

Comment:
These comments should say `get/setCollidable`. I suspect this is left over from a rename.